robot/packages/cloud_package_manager.go

```diff
@@ -299,23 +301,30 @@ func sanitizeURLForLogs(u string) string {
 	return parsed.String()
 }
 
-func (m *cloudManager) loadFile(ctx context.Context, url string, p config.PackageConfig) error {
+func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config.PackageConfig) error {
 	// TODO(): validate integrity of directory.
-	if dirExists(m.localDataPath(p)) {
+	if dirExists(p.LocalDataDirectory(m.packagesDir)) {
 		m.logger.Debug("  Package already downloaded, skipping.")
 		return nil
 	}
 
+	if err := os.MkdirAll(p.LocalDataParentDirectory(m.packagesDir), 0o700); err != nil {
+		return err
+	}
+
 	// Force redownload of package archive.
 	if err := m.cleanup(p); err != nil {
 		m.logger.Debug(err)
 	}
-	if err := os.Remove(m.localNamedPath(p)); err != nil {
-		m.logger.Debug(err)
+
+	if p.Type == config.PackageTypeMlModel {
```

Comment:
[q] why we are only doing this for ml models? should we not be doing it for the module package too that is not symlinked?

Reply:
This is just removing top level symlinks ex: `packages/my-cool-ml-model`

Reply:
Lots more detail in https://viam.atlassian.net/browse/RSDK-4386